src/gwc/src/main/java/org/geoserver/gwc/layer/GeoServerTileLayer.java

```diff
@@ -1522,4 +1534,66 @@ private static String baseUrl() {
         // no HTTP request is in progress
         return null;
     }
+
+    @Override
+    public boolean supportsTileJSON() {
+        return getGridSubsetForSRS(SRS.getEPSG3857()) != null
+                || getGridSubsetForSRS(SRS.getEPSG900913()) != null;
+    }
+
+    @Override
+    public TileJSON getTileJSON() {
+        TileJSON tileJSON = new TileJSON();
+        tileJSON.setName(getName());
+        LayerMetaInformation metaInformation = getMetaInformation();
+        if (metaInformation != null) {
+            tileJSON.setDescription(metaInformation.getDescription());
+        }
+        BoundingBox wgs84Bounds = getBounds(SRS.getEPSG4326());
+        PublishedInfo publishedInfo = getPublishedInfo();
+        PublishedType type = publishedInfo.getType();
+        if (type == PublishedType.VECTOR) {
+            setVectorLayers(tileJSON, publishedInfo);
+        }
+
+        tileJSON.setBounds(
+                new double[] {
+                    wgs84Bounds.getMinX(),
+                    wgs84Bounds.getMinY(),
+                    wgs84Bounds.getMaxX(),
+                    wgs84Bounds.getMaxY()
+                });
+
+        return tileJSON;
+    }
+
+    private void setVectorLayers(TileJSON tileJSON, PublishedInfo publishedInfo) {
+        if (Proxy.isProxyClass(publishedInfo.getClass())) {
+            LayerInfo inner = (LayerInfo) ModificationProxy.unwrap(publishedInfo);
+            ResourceInfo resource = inner.getResource();
+            if (resource instanceof FeatureTypeInfo) {
```

Comment:
Should have a branch for layer groups too, expand the group, if fully made of vector layers, add the vector layer info for each one of them.